photon-ml/src/integTest/scala/com/linkedin/photon/ml/GameModelGenerator.scala

```diff
@@ -0,0 +1,102 @@
+ * not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linkedin.photon.ml
+
+import org.apache.log4j.Logger
+import org.apache.spark.SparkContext
+
+import com.linkedin.photon.ml.TaskType.TaskType
+import com.linkedin.photon.ml.model.{Coefficients, FixedEffectModel, GAMEModel, RandomEffectModel}
+import com.linkedin.photon.ml.supervised.classification.LogisticRegressionModel
+import com.linkedin.photon.ml.supervised.regression.LinearRegressionModel
+
+/**
+ * A generator of Game models.
+ *
+ * The models generated are random, within certain parameters that the user can specify (e.g., number of random effects,
+ * number of features in each random effect, ...).
+ */
+class GameModelGenerator {
+
+  private val log : Logger = Logger.getLogger(classOf[GameModelGenerator])
+  private val rng = scala.util.Random
+
+  private def randInt(M: Int) = 1 + math.abs(rng.nextInt) % M
+
+  private def randSparseVector(M: Int) = {
+    val indices = (0 until M).map { _ => randInt(M) }.toSet.toArray.sorted
+    val values = indices.indices.map { _ => rng.nextDouble }
+    (indices, values)
+  }
+
+  /**
+   * Generate a random Game model according to parameters configured by the user.
+   *
+   * @param sc The Spark context
+   * @param taskType The type of model (see TaskType for supported types)
+   * @param NRE The number of random effects in the generated model
+   * @param maxNF The maximum number of features in each random effect. This is only a maximum, because the actual
+   *              number will be drawn at random from the uniform distribution over [1, maxNF).
+   * @param maxNSubs The maximum number of sub-models per random effect. This is again only a maximum, the actual
+   *                 number being drawn at random from the uniform distribution over [1, maxNSubs).
+   * @return a well formed Game model.
+   */
+  def generateGameModel(
+      sc: SparkContext,
+      taskType: TaskType,
+      NRE: Int = 100,
+      maxNF: Int = 100,
+      maxNSubs: Int = 100): GAMEModel = {
+
+    log.info(
+      s"Creating a $taskType Game model " +
+        s"with $NRE random effects, " +
+        s"at most $maxNF features per random effect " +
+        s"and at most $maxNSubs sub-models per random effect")
+
+    val numFeatures = (0 until NRE).map { i => (s"Effect$i", randInt(maxNF)) }.toMap
+
+    val model = taskType match {
+      case TaskType.LINEAR_REGRESSION => LinearRegressionModel.apply _
+      case TaskType.LOGISTIC_REGRESSION => LogisticRegressionModel.apply _
```

Comment:
This is taken from the converter for Laser, which doesn't handle Poisson regression yet. But you are right, I should add it. No reason we can't have Poisson regression here.